### PR TITLE
[CELEBORN-1190] Apply error prone patch and suppress some problems

### PR DESCRIPTION
**Pull request by cxzl25**
### What changes were proposed in this pull request?
1.  Fix MissingOverride, DefaultCharset, UnnecessaryParentheses Rule
2. Exclude generated sources, FutureReturnValueIgnored, TypeParameterUnusedInFormals, UnusedVariable

### Why are the changes needed?
```
./build/make-distribution.sh --release
```
We get a lot of WARNINGs.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
GA

### CODE REVIEW

**Comment by pan3793** [comment]
Emm... I think at least we can not fix all `MissingOverride` cases while supporting multiple Spark/Flink versions.

File: client-spark/spark-3-columnar-shuffle/src/main/java/org/apache/spark/sql/execution/columnar/CelebornColumnDictionary.java [hunk 1]
```diff
@@ -58,6 +60,6 @@ public double decodeToDouble(int id) {
 
   @Override
   public byte[] decodeToBinary(int id) {
-    return stringDictionary[id].getBytes();
+    return stringDictionary[id].getBytes(UTF_8);
```
Comment (pan3793) [inline client-spark/spark-3-columnar-shuffle/src/main/java/org/apache/spark/sql/execution/columnar/CelebornColumnDictionary.java:63]: Avoid static import unless it has real improvement for code style
```suggestion
    return stringDictionary[id].getBytes(StandardCharsets.UTF_8);
```

**Comment by cxzl25** [comment]
> I think at least we can not fix all `MissingOverride` cases while supporting multiple Spark/Flink versions.

So I kept some methods without adding `@Override`, which is why GA passed.

**Comment by pan3793** [comment]
cc @cfmcgrady, consider including this patch into 0.4.0 (or at least branch-0.4) to reduce the potential conflicts of future cherry-picking

**Comment by cfmcgrady** [comment]
thanks, merging to main(v0.5.0)/branch-0.4(v0.4.0).

**Pull request closed without merging**